addons/product/models/product_template.py

```diff
@@ -174,6 +175,17 @@ def _compute_currency_id(self):
     def _compute_cost_currency_id(self):
         self.cost_currency_id = self.env.company.currency_id.id
 
+    @api.depends('type')
+    def _compute_uom_category_ids(self):
+        for p in self:
+            if p.type == 'service':
+                p.uom_categ_ids = self.env.ref('uom.uom_categ_wtime').ids
+                # Change the default uom from 'Unit' to Hour if the current category_id is not wtime
+                if p.uom_id.category_id.ids != p.uom_categ_ids.ids:
```

Comment:
Use
```suggestion
                if p.uom_id.category_id.id not in p.uom_categ_ids.ids:
```
instead to improve the readability of the code

Reply:
`if p.uom_id.category_id not in p.uom_categ_ids:`
you can remove `.id` and `.ids`